File: backend/app/admin/service/internal/service/file.go
package service

import (
	"context"

	"github.com/go-kratos/kratos/v2/log"
	"github.com/tx7do/go-utils/trans"
	"google.golang.org/protobuf/types/known/emptypb"

	"kratos-admin/app/admin/service/internal/data"

	pagination "github.com/tx7do/kratos-bootstrap/api/gen/go/pagination/v1"
	adminV1 "kratos-admin/api/gen/go/admin/service/v1"
	fileV1 "kratos-admin/api/gen/go/file/service/v1"

	"kratos-admin/pkg/middleware/auth"
)

type FileService struct {
	adminV1.FileServiceHTTPServer

	log *log.Helper

	uc *data.FileRepo
}

func NewFileService(uc *data.FileRepo, logger log.Logger) *FileService {
	l := log.NewHelper(log.With(logger, "module", "file/service/admin-service"))
	return &FileService{
		log: l,
		uc:  uc,
	}
}

func (s *FileService) List(ctx context.Context, req *pagination.PagingRequest) (*fileV1.ListFileResponse, error) {
	return s.uc.List(ctx, req)
}

func (s *FileService) Get(ctx context.Context, req *fileV1.GetFileRequest) (*fileV1.File, error) {
	return s.uc.Get(ctx, req)
}

func (s *FileService) Create(ctx context.Context, req *fileV1.CreateFileRequest) (*emptypb.Empty, error) {
	if req.Data == nil {
		return nil, adminV1.ErrorBadRequest("错误的参数")
	}

	// 获取操作人信息
	operator, err := auth.FromContext(ctx)
	if err != nil {
		return &emptypb.Empty{}, err
	}

	req.Data.CreateBy = trans.Ptr(operator.UserId)

	if err = s.uc.Create(ctx, req); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *FileService) Update(ctx context.Context, req *fileV1.UpdateFileRequest) (*emptypb.Empty, error) {
	if req.Data == nil {
		return nil, adminV1.ErrorBadRequest("错误的参数")
	}

	// 获取操作人信息
	operator, err := auth.FromContext(ctx)
	if err != nil {
		return &emptypb.Empty{}, err
	}

	req.Data.UpdateBy = trans.Ptr(operator.UserId)

	if err = s.uc.Update(ctx, req); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *FileService) Delete(ctx context.Context, req *fileV1.DeleteFileRequest) (*emptypb.Empty, error) {
	if err := s.uc.Delete(ctx, req); err != nil {
		return nil, err
	}
	return &emptypb.Empty{}, nil
}

File: frontend/apps/admin/src/rpc/service/file.ts
import type { FileService } from '#/rpc/api/admin/service/v1/i_file.pb';
import type {
  CreateFileRequest,
  DeleteFileRequest,
  File,
  GetFileRequest,
  ListFileResponse,
  UpdateFileRequest,
} from '#/rpc/api/file/service/v1/file.pb';
import type { Empty } from '#/rpc/api/google/protobuf/empty.pb';
import type { PagingRequest } from '#/rpc/api/pagination/v1/pagination.pb';

import { requestClient } from '#/rpc/request';

/** 文件管理服务 */
class FileServiceImpl implements FileService {
  async CreateFile(request: CreateFileRequest): Promise<Empty> {
    return await requestClient.post<Empty>('/files', request);
  }

  async DeleteFile(request: DeleteFileRequest): Promise<Empty> {
    return await requestClient.delete<Empty>(`/files/${request.id}`);
  }

  async GetFile(request: GetFileRequest): Promise<File> {
    return await requestClient.get<File>(`/files/${request.id}`);
  }

  async ListFile(request: PagingRequest): Promise<ListFileResponse> {
    return await requestClient.get<ListFileResponse>('/files', {
      params: request,
    });
  }

  async UpdateFile(request: UpdateFileRequest): Promise<Empty> {
    const id = request.data?.id;

    console.log('UpdateFile', request.data);
    if (request.data !== null && request.data !== undefined) {
      request.data.id = undefined;
    }
    return await requestClient.put<Empty>(`/files/${id}`, request);
  }
}

export const defFileService = new FileServiceImpl();
